src/utils/settingsValidation/logger/commons.ts

```diff
@@ -0,0 +1,24 @@
+
+import { LogLevels, isLogLevelString } from '../../../logger';
+import { LogLevel } from '../../../types';
+
+/**
+ * Returns the LogLevel for the given debugValue or undefined if it is invalid,
+ * i.e., if the debugValue is not a boolean or LogLevel string.
+ *
+ * @param debugValue debug value at config
+ * @returns LogLevel of the given debugValue
+ */
+export function getLogLevel(debugValue: unknown): LogLevel | undefined {
+  if (typeof debugValue === 'boolean') {
+    if (debugValue) {
+      return LogLevels.DEBUG;
+    } else {
+      return LogLevels.NONE;
+    }
+  } else if (typeof debugValue === 'string' && isLogLevelString(debugValue)) {
+    return debugValue;
+  } else {
+    return undefined;
```

Comment:
Is there an scenario where `undefined` is a valid case? If not, maybe the 'last resort' return should be the `NONE` logLevel (I may be missing a lot of context, but usually you try to avoid returning an `undefined` value).

Reply:
returning undefined is the way the `getLogLevel` "validator" tell us that the provided input (`debugValue` param) is invalid. 
Other validators, in the case of invalid input, either: (a) logs a message and return a default value, or (b) throws an error. In the `getLogLevel` case, the former was not possible (it doesn't have the logger instance to log the error), and the later I think is overkill in terms of extra code.